pyorbital/tests/test_orbit_elements.py

```diff
@@ -0,0 +1,374 @@
+        (80.0, False),
+    ],
+)
+def test_is_retrograde_property(inclination_deg, expected):
+    """Test retrograde orbit detection based on inclination."""
+    tle = MockTLE()
+    tle.inclination = inclination_deg
+    orbit = OrbitElements(tle)
+    assert orbit.is_retrograde == expected
+
+
+@pytest.mark.parametrize("excentricity", [0.001, 0.01, 0.1, 0.5])
+def test_velocity_perigee_greater_than_apogee(excentricity):
+    """Ensure velocity at perigee is greater than at apogee for elliptical orbits."""
+    tle = MockTLE()
+    tle.excentricity = excentricity
+    orbit = OrbitElements(tle)
+    v_perigee = orbit.velocity_at_perigee()
+    v_apogee = orbit.velocity_at_apogee()
+    assert v_perigee > v_apogee
+
+
+@pytest.mark.parametrize("excentricity", [0.0, ECC_EPS / 10, ECC_EPS])
+def test_velocity_equal_for_circular_orbits(excentricity):
+    """Ensure velocity at perigee equals velocity at apogee for nearly circular orbits."""
+    tle = MockTLE()
+    tle.excentricity = excentricity
+    orbit = OrbitElements(tle)
+    v_perigee = orbit.velocity_at_perigee()
+    v_apogee = orbit.velocity_at_apogee()
+    assert v_perigee == pytest.approx(v_apogee, rel=1e-5)
+
+
+def test_high_eccentricity_limit():
+    """Test behavior near the upper eccentricity limit."""
+    tle = MockTLE()
+    tle.excentricity = ECC_LIMIT_HIGH
+    orbit = OrbitElements(tle)
+    assert orbit.semi_major_axis > 0
+    assert orbit.velocity_at_perigee() > orbit.velocity_at_apogee()
+
+
+def test_sgp4_unnormalization_value():
+    """Validate SGP4 un-normalization of mean motion and semi-major axis.
+
+    This test uses OrbitElements internal formulas for consistency.
+    """
+
+    class RefTLE(MockTLE):
+        """Reference TLE with fixed orbital parameters for precision testing."""
+
+        def __init__(self):
+            """Initialize reference TLE values for SGP4 unnormalization test."""
+            super().__init__()
+            self.inclination = 98.7408  # degrees
+            self.excentricity = 0.001140
+            self.mean_motion = 14.28634842  # rev/day
+            self.epoch = datetime.datetime(2200, 1, 1, 0, 0, 0)
+
+    tle = RefTLE()
+    orbit = OrbitElements(tle)
+
+    # Recompute reference values using the same formulas
+    n0_rad_per_min = tle.mean_motion * (2 * np.pi / XMNPDA)
+    a_1 = (XKE / n0_rad_per_min) ** (2.0 / 3)
+    delta_1 = (
+        (3 / 2.0)
+        * (CK2 / a_1**2)
+        * (
+            (3 * np.cos(np.deg2rad(tle.inclination)) ** 2 - 1)
+            / (1 - tle.excentricity**2) ** (3 / 2)
+        )
+    )
+    a_0 = a_1 * (1 - delta_1 / 3 - delta_1**2 - (134.0 / 81) * delta_1**3)
+    delta_0 = (
+        (3 / 2.0)
+        * (CK2 / a_0**2)
+        * (
+            (3 * np.cos(np.deg2rad(tle.inclination)) ** 2 - 1)
+            / (1 - tle.excentricity**2) ** (3 / 2)
+        )
+    )
+
+    expected_mean_motion = n0_rad_per_min / (1 + delta_0)
+    expected_semi_major_axis = a_0 / (1 - delta_0)
```

Comment:
Same here. Can we hardcoded the expected values? Otherwise we're just testing that the code in the test is the same as the code in the package and not the actual expected values. Sorry if I'm misunderstanding the code.